### PR TITLE
Singleton style tag

### LINKED ISSUE #2905: [feature]: Upgrade style-loader
**Is your feature request related to a problem? Please describe.**
It would be great to upgrade style-loader to `2.0.0` and leverage the loader options. Currently the version `~0.23.1` is used.

The main thing that would improve the style loading is the `injectType: singletonStyleTag` option, which would place the bundled styles into a single style tag.

**Describe the solution you'd like**
- Upgrade style-loader package to `~2.0.0`
- Add the following loader options to .css files:
```
{
                    loader: 'style-loader',
                    options: { injectType:'singletonStyleTag' }
},
```

**Additional context**

*Current result of the style-loader*
image

**Please let us know what packages this feature is in regards to:**
- [ ] `venia-concept`
- [ ] `venia-ui`
- [x] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by papiliond**
## Description
- Updated style-loader version from `~0.23.1` to `~2.0.0`.
- Use `injectType: singletonStyleTag` in production mode to avoid having a separate style tag for every css module.
- Use `injectType: styleTag` in development mode, because `singletonStyleTag` inject type does not support sourceMaps.

## Related Issue
Closes #2905 

### Verification Steps
1. Serve venia-concept in development mode. Open the site and check the `style` tags under the `head` tag. The css modules are transformed into many style tags.
2. Serve venia-concept in production mode, Open the site and check the `style` tags under the `head` tag, The css modules are transformed into a single style tag.

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by jimbo — DISMISSED** [review]
Looks good to me. Thanks for contributing!

Will put this in queue for QA.

File: packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getModuleRules.js [hunk 1]
```diff
@@ -96,13 +96,21 @@ getModuleRules.js = async ({
...
  */
-getModuleRules.css = async ({ paths, hasFlag }) => ({
+getModuleRules.css = async ({ mode, paths, hasFlag }) => ({
     test: /\.css$/,
     oneOf: [
         {
             test: [paths.src, ...hasFlag('cssModules')],
             use: [
-                'style-loader',
+                {
```
Comment (revanth0212) [inline packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getModuleRules.js:105]: @jcalcaben you may wanna update the docs with this new webpack config for style tags. A quick search for `style-loader` showed a couple of doc files mentioning it.

Comment (papiliond) [reply packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getModuleRules.js:105]: @revanth0212 I've updated the docs at css-modules configuration.

https://github.com/magento/pwa-studio/pull/2906/commits/47731764e7ff5103f4a5ada08e64b4787b715184

**Comment by sirugh** [comment]
Apologies. This fell through the cracks but I have moved it to the appropriate state on our board. Please try to resolve the conflicts and we can get this reviewed and then moved to QA!

**Review by supernova-at — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved.

**Review by dpatil-magento — APPROVED** [review]
Approved test commit.

**Comment by papiliond** [comment]
:label: partner-contribution

**Pull request merged**